assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -1639,13 +1640,18 @@ public void getSubmissionsZip(OutputStream out, String reference, String query)
                     List<AssignmentSubmission> submissions = new ArrayList<AssignmentSubmission>(submitters.values());
 
                     StringBuilder exceptionMessage = new StringBuilder();
-
+                    SortedIterator sortedIterator;
+                    if (assignmentUsesAnonymousGrading(assignment)){
+                        sortedIterator = new SortedIterator(submissions.iterator(), new AnonymousSubmissionComparator());
```

Comment:
This should cause some type of exception as there is a private no arg constructor, meaning should never instantiate this class using the noarg constructor.

Reply:
But this class does not have a private no arg constructor, it's the AssignmentSubmissionComparator one.

Reply:
thanks for the clarification!